main.cpp

```diff
@@ -126,384 +167,529 @@ struct CarWash
+    }
+    return powerCommand;
+}
+
+void GamingLaptop::LaptopBottom::inputCharacter(std::string character, bool malfunction)
+{
+    if (malfunction == false)
+    {
+        std::cout << character << std::endl;
+    }
+    else
+    {
+        std::cout << "Hello your keyboard has broken\n";
+    }
+    return;
+}
+
+bool GamingLaptop::LaptopBottom::powerWithBattery(int capacity, int consumption, float duration)
+{
+    bool powerState;
+    while (capacity > 0)
+    {
+        powerState = true;
+        capacity -= (consumption * duration);
+    }
+    powerState = false;
+    return powerState;
+}
+
+void GamingLaptop::launchGameA(std::string quality)
+{
+    std::cout << "Launching Game A at " << quality << " settings\n";
+    return;
+}
+
+int GamingLaptop::exportTestVideo(int quality, int videoLength)
+{
+    int exportDuration;
+    int gpuStrength = 0;
+    exportDuration = quality * videoLength * gpuStrength;
+    return exportDuration;
+}
+
+void connectPortToExternal(std::string type, std::string device)
+{
+    std::cout << "Device " << device << " detected and connected to " << type << " port.\n";
+    return;
+}
+
 struct PublicRestroomWithManagement
 {
-    // Average number of patrons using the toilet per week
     float averageUserNumber = 100.4f;
-    // Total number of toilets available in room
     int numberToiletsTotal = 5;
-    // Total number of urinals in room
     int numberUrinalsTotal = 10;
-    // Total number of sinks in room
     int numberSinksTotal = 7; 
-    // Average volume of water consumed per week by toilet facilities
     double waterUsagePerWeek = 70.0;
 
-    // Order by management to unclog toilets
     void unclogToilets(int numberCloggedToilets);
-    // Order by management to clean facilities
     void cleanFacilities(std::string facilityIssuesToAddress);
-    // Order by management to replenish soap at sinks
     void replenishSoap(float currentPercentageOfSoapLeft = 1.00f);
 };
-/*
-Thing 4) Military Trunk Communications Vehicle
-5 properties:
-    1) Number of men assigned to vehicle (int)
-    2) Fuel level percentage of vehicle (double)
-    3) Maximum height of antenna (int)
-    4) Configured frequency of radio systems (float)
-    5) Net power consumption of electronics (float)
-3 things it can do:
-    1) Change radio frequency
-    2) Deactivate comms systems during radio silence
-    3) Jack up antenna to specified height
- */
+
+void PublicRestroomWithManagement::unclogToilets(int totalToilets)
+{
+    std::cout << totalToilets << " toilets will be cleaned now.";
+    return;
+}
+
+void PublicRestroomWithManagement::cleanFacilities(std::string issue)
+{
+    std::cout << "Order to address " << issue << " sent to manpower.\n";
+    return;
+}
+
+void PublicRestroomWithManagement::replenishSoap(float percentageSoapLeft)
+{
+    float percentageToRefill = 100 * (1 - percentageSoapLeft);
+    std::cout << "Order sent to refill " << percentageToRefill << "% of soap.\n";
+    return;
+}
```

Comment:
Functions that don't return anything don't need a `return;` statement
- [ ] revise